src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -340,23 +340,25 @@ public class BigInteger extends Number implements Comparable<BigInteger> {
      *         two's-complement binary representation of a BigInteger.
      * @param  off the start offset of the binary representation.
      * @param  len the number of bytes to use.
-     * @throws NumberFormatException {@code val} is zero bytes long.
+     * @throws NumberFormatException {@code val} is zero bytes long,
+     *         or {@code len} is zero.
```

Comment:
I would be concerned about the compatibility impact on existing applications.
Though (very) uncommon, an existing application be using a non-zero length array and a len of zero to represent zero.
With this change it will throw an unexpected exception.

Reply:
A zero `len` is meant to indicate an empty range, whether the `val` array is empty or not.
So, from a logical perspective, both cases `val.length == 0` or `len == 0` should either both throw, or both should return `BigInteger` 0.
I choose the former, but the latter is a consistent choice as well.

What I find unpleasant is the current behavior: it throws on `val.length == 0` (even if `len == 0`), but curiously returns 0 when it is non-zero and `len == 0`. I find this rather inconsistent.

Reply:
The `len == 0` -> 0 behavior has been recovered.